File: scripts/test_responsive.js
const fs = require('fs/promises');
const path = require('path');
const puppeteer = require('puppeteer');

const projectRoot = path.resolve(__dirname, '..');
const screenshotDir = process.argv[2]
  ? path.resolve(process.cwd(), process.argv[2])
  : path.join(projectRoot, 'artifacts', 'responsive');

async function capturePage(page, options) {
  const { url, width, height, filename, waitForSelector } = options;

  await page.setViewport({ width, height });
  await page.goto(url, { waitUntil: 'networkidle0' });

  if (waitForSelector) {
    await page.waitForSelector(waitForSelector);
  }

  await page.screenshot({
    path: path.join(screenshotDir, filename),
    fullPage: true,
  });
}

async function run() {
  await fs.mkdir(screenshotDir, { recursive: true });

  const browser = await puppeteer.launch({ headless: true });

  try {
    const desktopPage = await browser.newPage();
    await capturePage(desktopPage, {
      url: 'http://127.0.0.1:8000/dashboard/',
      width: 1280,
      height: 800,
      filename: 'desktop_dashboard.png',
      waitForSelector: '.categories-grid',
    });

    await capturePage(desktopPage, {
      url: 'http://127.0.0.1:8000/expenses/',
      width: 1280,
      height: 800,
      filename: 'desktop_expenses.png',
      waitForSelector: '.desktop-form-grid',
    });

    const mobilePage = await browser.newPage();
    await capturePage(mobilePage, {
      url: 'http://127.0.0.1:8000/expenses/',
      width: 390,
      height: 844,
      filename: 'mobile_expenses.png',
      waitForSelector: '.page-title',
    });

    console.log(`Responsive screenshots saved to ${screenshotDir}`);
  } finally {
    await browser.close();
  }
}

run().catch((error) => {
  console.error(error);
  process.exit(1);
});

File: scripts/test_responsive.py
import asyncio
from playwright.async_api import async_playwright
import os
import time

async def capture_screenshots():
    # Setup paths
    base_dir = "/Users/muradbakirov/.gemini/antigravity/brain/05545778-74ee-4c3a-a780-60fb005e5c7a/"
    desktop_dashboard = os.path.join(base_dir, "desktop_dashboard_final.png")
    desktop_expenses = os.path.join(base_dir, "desktop_expenses_final.png")
    mobile_expenses = os.path.join(base_dir, "mobile_expenses_final.png")

    async with async_playwright() as p:
        browser = await p.chromium.launch(headless=True)
        
        # 1. Desktop Context
        desktop_context = await browser.new_context(
            viewport={'width': 1280, 'height': 800}
        )
        desktop_page = await desktop_context.new_page()
        
        # Dashboard Desktop
        await desktop_page.goto("http://127.0.0.1:8000/dashboard/")
        await desktop_page.wait_for_selector(".categories-grid")
        time.sleep(1) # wait for animations
        await desktop_page.screenshot(path=desktop_dashboard, full_page=True)
        print(f"Captured Desktop Dashboard: {desktop_dashboard}")
        
        # Expenses Desktop
        await desktop_page.goto("http://127.0.0.1:8000/expenses/")
        await desktop_page.wait_for_selector(".desktop-form-grid")
        time.sleep(1)
        await desktop_page.screenshot(path=desktop_expenses, full_page=True)
        print(f"Captured Desktop Expenses: {desktop_expenses}")
        await desktop_context.close()
        
        # 2. Mobile Context
        mobile_context = await browser.new_context(
            viewport={'width': 390, 'height': 844}
        )
        mobile_page = await mobile_context.new_page()
        
        # Expenses Mobile
        await mobile_page.goto("http://127.0.0.1:8000/expenses/")
        time.sleep(1)
        await mobile_page.screenshot(path=mobile_expenses, full_page=True)
        print(f"Captured Mobile Expenses: {mobile_expenses}")
        await mobile_context.close()
        
        await browser.close()

if __name__ == "__main__":
    asyncio.run(capture_screenshots())
